pkg/crc/cluster/cluster.go

```diff
@@ -510,3 +510,29 @@ func DeleteMCOLeaderLease(ctx context.Context, ocConfig oc.Config) error {
 	_, _, err := ocConfig.RunOcCommand("delete", "-A", "lease", "--all")
 	return err
 }
+
+func CheckCorePodsRunning(ctx context.Context, ocConfig oc.Config) error {
+	coreNameSpace := []string{"kube-system", "openshift-dns", "openshift-ingress", "openshift-ovn-kubernetes", "openshift-service-ca"}
+	waitForPods := func() error {
+		for _, namespace := range coreNameSpace {
+			if !podRunningForNamespace(ocConfig, namespace) {
+				logging.Debugf("Pods in %s namespace are not running", namespace)
+				return &errors.RetriableError{Err: fmt.Errorf("pods in %s namespace are not running", namespace)}
+			}
+		}
+		return nil
+	}
+	return errors.Retry(ctx, 2*time.Minute, waitForPods, 2*time.Second)
+}
+
+func podRunningForNamespace(ocConfig oc.Config, namespace string) bool {
+	stdin, stderr, err := ocConfig.WithFailFast().RunOcCommand("get", "pods", "-n", namespace, "--field-selector=status.phase!=Running")
```

Comment:
Why `!=Running`? I would have expected `=Running`?

Reply:
@cfergeau If I use `==Running` that means it will show all the running pods, what we want the pods which are not in Running phase in that specific namespace so we can reiterate in retry function.
```
$ kubectl get pod -n kube-system --field-selector=status.phase!=Running 
NAME                                       READY   STATUS    RESTARTS   AGE
csi-snapshot-controller-85cc4fd76b-xznzw   1/1     Pending   0          45h
```